module3-crud-and-data-models/r1-schema-and-data-models/README.md

```diff
@@ -0,0 +1,219 @@
+
+```js
+router.get("/dishes", async (_, res) => {
+  // here, we are using the find() to return all dishes from our MongoDB database
+  // find() will return an array of dish objects
+  const posts = await DishModel.find();
+  res.json(posts);
+});
+```
+
+#### Update
+
+PUT is the HTTP method used for the CRUD operation, Update.
+So if the price of Avocado Toast has gone up, we should go into the database and update that information using PUT.
+
+Note: Use `PUT` when you want to modify a singular resource which is already a part of resources collection. `PUT` replaces the resource with the data you send in its entirety.
+
+Request:
+
+`PUT http://www.myrestaurant.com/dishes/:id`
+
+We will need to send the new dish data too as an object.
+
+For example
+
+```js
+{
+  title: "New dish title",
+  image: "www.food.com/dish.jpg",
+  tags: ["simple", "classic"]
+}
+```
+
+Note that when using Mongoose you don't have to send all the object fields, you can only send the ones that got changed.
+
+For example, we can only send the title.
+
+```js
+{
+  title: "New dish title",
+}
+```
+
+Respectively, our express route for both these examples will be
+
+```js
+router.put("/dishes/:id", async (req, res) => {
+  // we need the dish id in order to tell Mongoose which dish to update in our database
+  const { id } = req.params;
+  try {
+    // the findByIdAndUpdate() will make sure the dish exist before updating it
+    const updatedDish = await DishModel.findByIdAndUpdate(id, req.body, {
+      // the { new: true } here is just to tell Mongoose we want the newely updated dish back
+      new: true,
+    });
+    // we return the updated dish as a JSON object to the user
+    res.json(updatedDish);
+  } catch (err) {
+    res.status(422).json({ message: err.message });
+  }
+});
+```
+
+#### Delete
+
+The CRUD operation Delete corresponds to the HTTP method DELETE. It is used to remove a resource from the system.
+Let’s say that the world avocado shortage has reached a critical point, and we can no longer afford to serve this modern delicacy at all. We should go into the database and delete the item that corresponds to **Avocado Toast**, which we know has an `id` of 1223.
+
+Request:
+
+`DELETE http://www.myrestaurant.com/dishes/:id`
+
+The express route for this deleting request can be
+
+```js
+router.delete("/dishes/:id", async (req, res) => {
+  const { id } = req.params;
+  try {
+    const dish = await PostModel.findByIdAndDelete(id);
+    // the status code is 204 because we are not returning data or entities
+    return res.status(204).json({ success: true });
+  } catch (err) {
+    res.status(422).json({ message: err.message });
+  }
+});
+```
+
+## Resources
+
+- [What is REST](https://restfulapi.net/)
+- [See this](https://www.educative.io/blog/crud-operations#what) for more information on how CRUD operations are performed directly on `SQL` database.
+- [database schemas](https://www.educative.io/blog/what-are-database-schemas-examples#types)
```

Comment:
Ideally these links should be those you have referred or copy/pasted from for the reading and are not required to be opened and ready by students. If you want to recommend any of these as a further reading, then mention them where relevant in the reading content as an inline link.